### ISSUE TITLE
Linked searches from record view:  Author

### ISSUE THREAD

**Issue by jvine** [sul-dlss/SearchWorks#1430]
Technically the same task as Subjects: for each Author in the record view, link to a fielded search for that Author name, in quotes. 

But, issues. 

Authors with roles and/or affiliations after their name - search fails if these terms are included.
https://sw-webapp-sandbox-f.stanford.edu/article/edsswe__edsswe_oai_DiVA_org_liu_134775
Sovacool, Benjamin K., **Author**
Linköpings universitet, Institutionen för tema, Tema Miljöförändring, **Originator**

Native UI sometimes has authors with email address, and/or links to author affiliations elsewhere in the record:
Barua, Prabal1 prabalims@gmail.com
Rahman, Syed Hafizur2 hafizsr@gmail.com
https://stanford.idm.oclc.org/login?url=http://search.ebscohost.com/login.aspx?direct=true&db=bth&AN=121364119&site=eds-live
but these don't seem to be in our results (maybe that's good?)
https://sw-webapp-sandbox-f.stanford.edu/article/bah__121364119

**Comment by drh-stanford** [comment]
@jvine do we have a list of relator terms that we can strip out programatically? We're not getting structured data for those authors

```
AUTHORS: ["Sovacool, Benjamin K., Author", "Linnér, Björn-Ola, Author", "Klein, Richard J. T., Author", "Linköpings universitet, Institutionen för tema, Tema Miljöförändring, Originator", "Linköpings universitet, Filosofiska fakulteten, Originator", "Linköpings universitet, Centrum för klimatpolitisk forskning, CSPR, Originator"]
```

In that example for affiliations, we're not getting any structured metadata for the authors:

```
AUTHORS: ["Barua, Prabal", "Rahman, Syed Hafizur"]
```

**Comment by jvine** [comment]
Need to ask EBSCO about relator terms - might depend on their metadata source.

**Comment by drh-stanford** [comment]
See https://github.com/ebsco/edsapi-ruby/issues/51 for the question of how to identify the relator terms

**Comment by drh-stanford** [comment]
Re: email addresses. The EDS gem has `eds_authors_composed` that has this metadata (that we're not using):

```html
<searchLink fieldCode="AR" term="%22Barua%2C+Prabal%22">Barua, Prabal</searchLink><relatesTo>1</relatesTo><i> prabalims@gmail.com</i><br />
<searchLink fieldCode="AR" term="%22Rahman%2C+Syed+Hafizur%22">Rahman, Syed Hafizur</searchLink><relatesTo>2</relatesTo><i> hafizsr@gmail.com</i>
```

**Comment by drh-stanford** [comment]
Re: index view. I'm assuming we want to remove the relator terms from the index view? Like so:

![screen shot 2017-07-18 at 3 47 04 pm](https://user-images.githubusercontent.com/1861171/28343099-7657d4ec-6bd0-11e7-9225-acd8d22dc534.png)

But include them in the show page, like so:

![screen shot 2017-07-18 at 3 48 10 pm](https://user-images.githubusercontent.com/1861171/28343111-8481a3a4-6bd0-11e7-93aa-bd1426468127.png)

**Comment by efrierson** [comment]
The EDS index includes metadata from over 20K different providers.  We do a _lot_ of normalization to populate the EDS record schema with the varying types of data we get from our partners, but this issue is an example of how you still might see variation in the content of the response.

Bill and I discussed this - I'm wondering if it would be appropriate to allow you to configure the gem to tell it what to do in the different types of scenarios you'll encounter with the data.  For example, rather than stripping our <searchLink> tags entirely and then having to grapple with metadata that might or might NOT include relator terms like "Author" and "Contributor", instead, use the <searchLinks> to identify the appropriate "linkable" portions of the field.

We could send back <a> tags, replacing the <searchLinks> with appropriate URLs to issue the desired search.  For example, if this _would have been_ the response:

<searchLink fieldcode="MM" term="%22Genetic+Testing+utilization%22">Genetic Testing utilization</searchLink>

We could return this instead:

<a href="/article?utf8=%E2%9C%93&format=html&search_field=search&q=MM+%22Genetic+Testing+utilization%22">Genetic Testing utilization</a>

All this would require is a patterned URL in a configuration to let us know how to build the href link.  

Same goes for <epLink>.

For <relatesTo>, we could return <sup>, or you could simply style the relatesTo tag to superscript, or we could strip out the tag entirely.

In short, it sounds like the functionality you all want (clickable authors, subject headings, research starters links) is _sort of_ in the EDS API already, but our Gem hasn't been handing that back to Blacklight in a way that's easy to use.  Rather than stripping all of our markup entirely and trying to re-engineer the functionality with potentially messy metadata, rely on the gem to take the best elements of the mark up and translate it to something Blacklight can display to achieve those goals.

**Comment by drh-stanford** [comment]
@efrierson @bmckinney Having a structured data response with the `searchLink` elements would probably be workable for us. The trick for us would be knowing which metadata fields would have the structured data. There's other metadata that has vanilla HTML markup, like `eds_abstract` -- Rails by default shows the tags to the user unless we mark that metadata as "`html_safe`". So, know which fields may have structured markup becomes important for us.

As for the current examples, right now, in the example above (`edsswe__edsswe_oai_DiVA_org_liu_134775`), we're getting the following for the composed authors (that is, no `searchLink` elements):

```
"eds_authors_composed": "Sovacool, Benjamin K., Author<br />Linnér, Björn-Ola, Author<br />Klein, Richard J. T., Author<br />Linköpings universitet, Institutionen för tema, Tema Miljöförändring, Originator<br />Linköpings universitet, Filosofiska fakulteten, Originator<br />Linköpings universitet, Centrum för klimatpolitisk forskning, CSPR, Originator"
```

whereas for the second example above (`bah__121364119 `) we are getting `searchLink` elements:

```
"eds_authors_composed": "<searchLink fieldCode=\"AR\" term=\"%22Barua%2C+Prabal%22\">Barua, Prabal</searchLink><relatesTo>1</relatesTo><i> prabalims@gmail.com</i><br /><searchLink fieldCode=\"AR\" term=\"%22Rahman%2C+Syed+Hafizur%22\">Rahman, Syed Hafizur</searchLink><relatesTo>2</relatesTo><i> hafizsr@gmail.com</i>",
```

**Comment by drh-stanford** [comment]
Blocked by https://github.com/ebsco/edsapi-ruby/issues/51

**Comment by saseestone** [comment]
Can't tell if this is still an issue.  Would need to be investigated further.
